crates/re_arrow_store/src/store_write.rs

```diff
@@ -994,42 +934,28 @@ impl PersistentComponentTable {
         }
     }
 
-    /// Pushes `rows_single` to the end of the bucket, returning the _global_ `RowIndex` of the
+    /// Pushes `cell` to the end of the bucket, returning the _global_ `RowIndex` of the
     /// freshly added row.
-    ///
-    /// `rows_single` must be a unit-length list of arrays of structs,
-    /// i.e. `ListArray<StructArray>`:
-    /// - the list layer corresponds to the different rows (always unit-length for now),
-    /// - the array layer corresponds to the different instances within that single row,
-    /// - and finally the struct layer holds the components themselves.
-    /// E.g.:
-    /// ```text
-    /// [[{x: 8.687487, y: 1.9590926}, {x: 2.0559108, y: 0.1494348}, {x: 7.09219, y: 0.9616637}]]
-    /// ```
-    //
-    // TODO(#589): support for batched row component insertions
-    pub fn push(&mut self, rows_single: &dyn Array) -> RowIndex {
+    pub fn push_cell(&mut self, cell: &DataCell) -> RowIndex {
         crate::profile_function!();
 
         debug_assert!(
-            ListArray::<i32>::get_child_type(rows_single.data_type()) == &self.datatype,
+            cell.datatype() == &self.datatype,
             "trying to insert data of the wrong datatype in a component table, \
                 expected {:?}, got {:?}",
             &self.datatype,
-            ListArray::<i32>::get_child_type(rows_single.data_type()),
-        );
-        debug_assert!(
-            rows_single.len() == 1,
-            "batched row component insertions are not supported yet"
+            cell.datatype(),
         );
 
+        // TODO(cmc): don't use raw arrays
+        let values = cell.as_arrow_monolist();
+
         self.total_rows += 1;
         // Warning: this is surprisingly costly!
-        self.total_size_bytes +=
-            arrow2::compute::aggregate::estimated_bytes_size(rows_single) as u64;
+        self.total_size_bytes += arrow2::compute::aggregate::estimated_bytes_size(&*values) as u64;
 
         // TODO(#589): support for non-unit-length chunks
-        self.chunks.push(rows_single.to_boxed()); // shallow
+        self.chunks.push(values); // shallow
```

Comment:
I think the comment is outdated
```suggestion
        self.chunks.push(values);
```